lib/wasi/src/state/mod.rs

```diff
@@ -672,6 +682,14 @@ impl WasiFs {
         Ok(cur_inode)
     }
 
+    /// Splits a path into the first preopened directory that is a parent of it
```

Comment:
Optional: is there guaranteed to be one? (Is it root? Maybe that doesn't belong documented here but somewhere else.)